README.md

```diff
@@ -1,2 +1,276 @@
+
+      ```bash
+      zephyr/scripts/twister \
+        -T example-application/zephyr/projects/zxa_example_project
+      ```
+
+1. Build and run the all projects (including projects and tests)
+   with code coverage using Zephyr's `twister script`
+
+      ```bash
+      zephyr/scripts/twister --coverage \
+        -T example-application/zephyr
+      ```
+
+### macOS
+
+1. Build projects directly from the command line
+
+      ```bash
+      cd ~/zxa
+      west build -p auto -b reel_board \
+        example-application/zephyr/projects/zxa_example_project
+      ```
+
+1. Run projects from the command line
+
+      ```bash
+      ./build/example-application/zephyr/projects/zxa_example_project/native_posix/zephyr/zephyr.exe
+      ```
+
+1. Build and run projects using Zephyr's `twister script`
+
+      ```bash
+      zephyr/scripts/twister \
+        -T example-application/zephyr/projects/zxa_example_project
+      ```
+
+1. Build and run the all projects (including projects and tests)
+   with code coverage using Zephyr's `twister script`
+
+      ```bash
+      zephyr/scripts/twister --coverage \
+        -T example-application/zephyr
+      ```
+
+### Windows
+
+1. Build projects directly from the command line
+
+      ```bat
+      cd %HOMEPATH%\zxa
+      west build -p auto -b native_posix \
+        example-application\zephyr\projects\zxa_example_project
+      ```
+
+1. Run projects from the command line
+
+      ```bat
+      .\build\example-application\zephyr\projects\zxa_example_project\native_posix\zephyr\zephyr.exe
+      ```
+
+1. Build and run projects using Zephyr's `twister script`
+
+      ```bat
+      zephyr\scripts\twister \
+        -T example-application\zephyr\projects\zxa_example_project
+      ```
+
+1. Build and run the all projects (including projects and tests)
+   with code coverage using Zephyr's `twister script`
+
+      ```bat
+      zephyr\scripts\twister --coverage \
+        -T example-application\zephyr
+      ```
+
+
+# Module Contribution Guidelines
+
+## Design Principles
+
+The ZXA module is designed around the following principles:
+
+- Keep all Zephyr-specific content within the (required) `zephyr` folder.
+
+  - This enables adding the example on to an existing repository.
+
+- Follow the Zephyr repository structure, mechanisms, and rules
+  whenever possible.
+
+  - This minimizes the variance from what users and developers expect,
+    simplifies maintenance, and reuses existing solutions.
+
+- Extend the Zephyr repository structure to support an arbitrary number
+  of applications and project configurations.
+
+  - The `projects` sub-tree identifies where non-test, non-sample projects
+    are expected to be found.  The `projects` folder is tied into the
+    Zephyr CI system through `module.yml`.
+
+- Maximize the amount of code automatically verifiable through Zephyr CI.
+
+- Minimize risk of conflicts in global namespaces through use of the
```

Comment:
When you make this statement you are taking a decision for the user. Some users might like prefixing with a global namespace, some others may not. Collision is a risk, but not having the namespace has advantages as well, for example better integration with existing subsystems if you extend them. By default we should offer the simplest scenario, and optionally document the possibility of prefixing everything. In general in Zephyr we do not prefix with a global namespace at all, besides the particular subsystem (`k_`, `bt_`, etc), so it follows that it is more consistent to just use names that are representative of the particular task they perform. While it is true that you could infer that prefixing the application itself follows this principle, this is not the case in our zephyr samples themselves, and adding the prefix goes against the simplicity principle: The application is the main entry point to the code, and thus users should be free to avoid namespaces altogether. I am happy to document that namespacing can be an interesting option for some, but it is certainly not required.

Reply:
This is only an example. Users are always free to follow it or not. Users are free to use it or not. No decisions have been taken away from the user (in fact, they can't be taken away). Rather, users are made aware of the risks and can easily strip out the prefixes along with unneeded functionality. Providing a rich implementation and instructions for how to strip out content is easier to use than following instructions on how to add content. This is part of the reason I liked calling this an 'example' rather than a 'sample'.  Using different terms indicates they mean different things.  It sounds like you want a `sample-application` repo to go along with the `example-application` repo.

Yes, I am aware that Zephyr does not prefix with a global namespace, which increases the risk and work of migrating my existing solution (and is a barrier to adoption).  I don't want to add to the existing risk especially as it propagates into user repositories.

Yes, the application is the entry point into the code.  More specifically, `main()` is the entry point to the code, and no one is  intentionally proposing changing the name of the function or restricting the containing file.  We can clarify that no code directly beneath `prj.conf` is impacted (because it shouldn't be shared).

Reply:
Since this is under the headings Module Contribution Guidelines, Design Principles this clearly does not apply to a user's copy of this repository, so it is not taking any decision from the user.  Rather it is a constraint on contributions back to *this* repository.